p2p-tests/lib/p2p.ts

```diff
@@ -104,66 +86,192 @@ export class LinePrefix extends stream.Transform {
   }
 }
 
-export function initPeer(namespace: string): void {
-  const radHome = workspacePath([namespace]);
-  fs.removeSync(radHome);
-  fs.mkdirsSync(radHome);
-
-  execa.sync(path.join(BIN_PATH, "radicle-proxy-init"), [
-    namespace,
-    "--key-passphrase",
-    namespace,
-    "--rad-home",
-    workspacePath([namespace]),
-  ]);
+interface RadicleProxyParams {
+  namespace: string;
+  ipAddress: string;
+  seed: string;
+  dataPath: string;
 }
 
-export function startPeer(
-  namespace: string,
-  ip: string,
-  args: string[]
-): ProxyClient.ProxyClient {
-  initPeer(namespace);
-
-  const childProcess = spawnInNamespace(
-    namespace,
-    [
-      path.join(BIN_PATH, "radicle-proxy"),
-      "--peer-listen",
-      `${ip}:8776`,
-      "--http-listen",
-      `${ip}:17246`,
-      "--skip-remote-helper-install",
-      "--unsafe-fast-keystore",
-      "--insecure-http-api",
-      "--dev-log",
-      ...args,
-    ],
-    {
-      RAD_HOME: workspacePath([namespace]),
+export class RadicleProxy {
+  public namespace: string;
+  public ipAddress: string;
+  public seed: string;
+  public dataPath: string;
+
+  public checkoutPath: string;
+  public passphrase: string;
+  public peerId: string;
+  public identityUrn: string;
+  public proxyClient: ProxyClient.ProxyClient;
+  public radHome: string;
+
+  public childProcess: execa.ExecaChildProcess | undefined = undefined;
+
+  constructor({ namespace, ipAddress, seed, dataPath }: RadicleProxyParams) {
+    this.dataPath = dataPath;
+    this.radHome = path.join(dataPath, `${namespace}-rad-home`);
+    this.namespace = namespace;
+    this.ipAddress = ipAddress;
+    this.seed = seed;
+
+    fs.mkdirsSync(this.radHome);
+
+    this.checkoutPath = path.join();
+    this.passphrase = namespace;
+
+    const initResult = JSON.parse(
+      execa.sync(path.join(BIN_PATH, "radicle-proxy-init"), [
+        this.namespace,
+        "--key-passphrase",
+        this.passphrase,
+        "--rad-home",
+        this.radHome,
+      ]).stdout
+    );
+
+    this.identityUrn = initResult.identityUrn;
+    this.peerId = initResult.peerId;
+
+    this.proxyClient = new ProxyClient.ProxyClient(
+      `http://${this.ipAddress}:17246`
+    );
+  }
+
+  start(): void {
```

Comment:
Could you add the appropriate public/private modifiers here and on all other methods? This should really be enforced by eslint.

Reply:
9993de28f197f8580c3a1919b1517321c4bc95bb